crates/compiler/load_internal/src/file.rs

```diff
@@ -192,7 +190,13 @@ fn start_phase<'a>(
                 // parse the file
                 let header = state.module_cache.headers.remove(&module_id).unwrap();
 
-                BuildTask::Parse { header }
+                BuildTask::Parse {
+                    header,
+                    // Provide mutexes of ModuleIds and IdentIds by module,
+                    // so other modules can populate them as they load.
```

Comment:
I'd either remove this comment entirely or put it on the type definition. It's hard to find otherwise.

Reply:
Yeah, this seems pretty self-descriptive.